### ISSUE TITLE
Question about finetune target

### ISSUE THREAD

**Issue by ZwhElliott** [FlagOpen/FlagEmbedding#935]
Hi, I've given fine-tuning the bge-m3 a shot, and the results have been fantastic.
 However, while going through the source code, I noticed that the target is auto-generated using the code below.

https://github.com/FlagOpen/FlagEmbedding/blob/1470fc6f325bd33e0c3e568b83ae616714f0ffef/FlagEmbedding/baai_general_embedding/finetune/modeling.py#L95-L101

I'm curious as to how the target label is auto-generated in this manner. The model uses dot product similarity to calculate similarity, implying that a higher score denotes greater similarity. 'Arange' method generates a tensor along the lines of [0,1,2,3...] as the target. Does this imply that the target generated this way is linked to the batch position of the query? Does a later position indicate more similarity?

Thank you very much.
